### PR TITLE
fix: check if user is blocked on profile popup

### PR DESCRIPTION
**Pull request by hydr063n**
This PR is a response to @emizzle's suggested change in PR #1431 . It checks if a user is blocked before exposing certain functionality to the user in a Profile popup. 

I've also added functionality to the StatusChatInput component that checks whether the chat recipient has been blocked and disabling it if the latter is true. The caveat is that the update to the `profileModel's blocked list` doesn't get picked up by the current active chat window. It only gets picked up when you find some way to reload the chat component eg switching chats/changing nickname

chat-inpout-blocked

### CODE REVIEW

File: ui/app/AppLayouts/Chat/components/ProfilePopup.qml [hunk 1]
```diff
@@ -371,15 +387,20 @@ ModalPopup {
...
             anchors.right: parent.right
-            anchors.rightMargin: addToContactsButton.width + 32
+            anchors.rightMargin: isBlocked ? 0 : addToContactsButton.width + 32
             btnColor: Style.current.lightRed
             btnBorderWidth: 1
             btnBorderColor: Style.current.grey
             textColor: Style.current.red
-            //% "Block User"
-            label: qsTrId("block-user")
+            label: isBlocked ? qsTr("Unblock User") : qsTr("Block User")
```
Comment (jrainville) [inline ui/app/AppLayouts/Chat/components/ProfilePopup.qml:395]: Could you put the `qsTr`s on two different lines for the context comments can be put above them correctly

**Review by jrainville — COMMENTED** [review]
Nice work.

For the problem for the input not updating when changing the block status, one way to fix it would be to add a signal in the profile view in Nim, then we can add a Connections that targets that view and that signal.

Here is an example of a NimQML signal: https://github.com/status-im/nim-status-client/blob/master/src/app/profile/view.nim#L358

You could call it `contactBlocked/contactUnblocked` and put the pubkey as the arg.

Then, in QML, you can do this: 
```qml
Connections {
    target: profileModel
    onContactBlocked {
        console.log(pubKey)
    }
}
```

File: ui/app/AppLayouts/Chat/components/ProfilePopup.qml (hunk 1 above)
Comment (hydr063n) [reply ui/app/AppLayouts/Chat/components/ProfilePopup.qml]: Done

**Comment by hydr063n** [comment]
Many thanks for the advice! The `Connections` helped me to update the relevant variable which makes the Chat input disable instantly after you block a user.

File: ui/app/AppLayouts/Chat/ChatColumn.qml [hunk 2]
```diff
@@ -68,6 +71,13 @@ StackLayout {
         txModalLoader.close()
     }
 
+    Connections {
+        target: profileModel
+        onContactListChanged: {
+            console.log("Contact list changed");
```
Comment (jrainville) [inline ui/app/AppLayouts/Chat/ChatColumn.qml:77]:
```suggestion
```

**Review by jrainville — COMMENTED** [review]
Nice works!
Just need to remove the leftover log

**Review by jrainville — APPROVED** [review]

File: ui/app/AppLayouts/Chat/ChatColumn.qml (hunk 2 above)
Comment (hydr063n) [reply ui/app/AppLayouts/Chat/ChatColumn.qml]: Damn I should pay better attention to these haha

**Pull request merged**